src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamespaceSymbol.cs

```diff
@@ -61,22 +66,31 @@ internal sealed override NamespaceExtent Extent
             AllowGenericEnumeration = false)]
         public sealed override ImmutableArray<Symbol> GetMembers()
         {
-            EnsureAllMembersLoaded();
+            if (_lazyFlattenedNamespacesAndTypes.IsDefault)
+            {
+                EnsureAllMembersLoaded();
+                ImmutableInterlocked.InterlockedExchange(ref _lazyFlattenedNamespacesAndTypes, calculateMembers());
+            }
 
-            var memberTypes = GetMemberTypesPrivate();
+            return StaticCast<Symbol>.From(_lazyFlattenedNamespacesAndTypes);
```

Comment:
Is ```StaticCast``` needed here?

Reply:
Nope, not after we changed the type of the collection